control-plane/plugin/src/resources/blockdevice.rs

```diff
@@ -0,0 +1,217 @@
+    }
+}
+
+// GetHeaderRow being trait for BlockDeviceUsable would return the Header Row for
+// BD returned from REST call with all set to false.
+impl GetHeaderRow for BlockDeviceUsable {
+    fn get_header_row(&self) -> Row {
+        (&*BLOCKDEVICE_HEADERS_USABLE).clone()
+    }
+}
+
+#[async_trait(?Send)]
+impl GetBlockDevices for BlockDevice {
+    type ID = NodeId;
+    async fn get_blockdevices(id: &Self::ID, all: &bool, output: &OutputFormat) {
+        let mut used_disks: Vec<String> = vec![];
+        match RestClient::client().pools_api().get_node_pools(id).await {
+            Ok(pools) => {
+                for pool in pools.into_body() {
+                    let mut normalized_disks: Vec<String> = pool
+                        .spec
+                        .unwrap()
+                        .disks
+                        .into_iter()
+                        .map(|disk| normalize_disk(disk.as_str()))
+                        .collect();
+                    used_disks.append(&mut normalized_disks);
+                }
+            }
+            Err(e) => {
+                println!("Failed to list blockdevices for node {} . Error {}", id, e)
+            }
+        }
+
+        match RestClient::client()
+            .block_devices_api()
+            .get_node_block_devices(id, Some(*all))
+            .await
+        {
+            Ok(blockdevices) => {
+                // Print table, json or yaml based on output format.
+                if *all {
+                    let bds: Vec<BlockDeviceAll> =
+                        transform(&used_disks, blockdevices.into_body(), true)
+                            .into_iter()
+                            .map(BlockDeviceAll)
+                            .collect();
+                    print_table(output, bds);
+                } else {
+                    let bds: Vec<BlockDeviceUsable> =
+                        transform(&used_disks, blockdevices.into_body(), false)
+                            .into_iter()
+                            .map(BlockDeviceUsable)
+                            .collect();
+                    print_table(output, bds);
+                };
+            }
+            Err(e) => {
+                println!("Failed to list blockdevices for node {} . Error {}", id, e)
+            }
+        }
+    }
+}
+
+fn transform(
+    used_disks: &[String],
+    blockdevices: Vec<openapi::models::BlockDevice>,
+    all: bool,
+) -> Vec<openapi::models::BlockDevice> {
+    let mut transformed_bds = vec![];
+    for mut bd in blockdevices {
+        let mut possible_values = vec![bd.devname.clone()];
+        for devlink in &bd.devlinks {
+            possible_values.push(devlink.clone());
+        }
+        if bd.available && check_available(used_disks, possible_values) {
+            bd.available = true;
+            transformed_bds.push(bd);
+        } else if all {
+            bd.available = false;
+            transformed_bds.push(bd);
```

Comment:
We don't need a new vector here right? we can return blockdevices if we use iter_mut on the for loop?

Reply:
Yeah.